message_ix/tests/reporting/test_computations.py

```diff
@@ -1,13 +1,75 @@
+from functools import partial
+
 import matplotlib
 import pandas as pd
 import pyam
+from dask.core import literal
+from genno.testing import random_qty
 from ixmp.reporting import Quantity
 
 from message_ix import Scenario
 from message_ix.reporting import Reporter, computations
 from message_ix.testing import SCENARIO
 
 
+def test_as_message_df(test_mp):
+    q = random_qty(dict(c=3, h=2, nl=5))
+    q.units = "kg"
+
+    args = (
+        literal("demand"),
+        dict(commodity="c", node="nl", time="h"),
+        dict(level="l", year=2022),
+    )
+
+    # Can be called directly
+    result0 = computations.as_message_df(q, *args, wrap=False)
+    assert not result0.isna().any().any()
+    # Values appear with the correct indices
+    assert (
```

Comment:
@gidden as requested, a check for pass-through of a specific value. Note that since `random_qty()` is used, the actual value changes with each test session, so we don't hard code it.

If nothing further, please ✅  and I'll merge.